shell/browser/api/electron_api_app.cc

```diff
@@ -921,6 +922,12 @@ void App::BrowserChildProcessCrashedOrKilled(
   if (!data.name.empty()) {
     details.Set("name", data.name);
   }
+  if (data.process_type == content::PROCESS_TYPE_UTILITY) {
+    base::ProcessId pid = data.GetProcess().Pid();
+    auto* utility_process_wrapper = GetAllUtilityProcessWrappers().Lookup(pid);
+    if (utility_process_wrapper)
+      utility_process_wrapper->Emit("exit", info.exit_code);
```

Comment:
I need to reconfirm this, but so far in my testing `UtilityProcessWrapper::OnServiceProcessDisconnected` was only called for graceful termination of the process, so this would never collide.

Reply:
In that case we need to properly unpin here too.

Reply:
https://github.com/electron/electron/pull/34980/commits/4a3a57985a314f72bf2afa0b8280d812aeec9c6a